echopype/calibrate/cal_params.py

```diff
@@ -1,155 +1,249 @@
-    "beamwidth_athwartship": "beamwidth_twoway_athwartship",
-    "z_et": "z_et",  # from default EK80 params
+EK80_DEFAULT_PARAMS = {
+    "impedance_transmit": 75,
+    "impedance_receive": 1000,
+    "receiver_sampling_frequency": {  # default full sampling frequency [Hz]
+        "default": 1500000,
+        "GPT": 500000,
+        "SBT": 50000,
+        "WBAT": 1500000,
+        "WBT TUBE": 1500000,
+        "WBT MINI": 1500000,
+        "WBT": 1500000,
+        "WBT HP": 187500,
+        "WBT LF": 93750,
+    },
 }
 
 
 # TODO: need a function (something like "_check_param_freq_dep")
 # to check user input cal_params and env_params
 
 
-def get_cal_params_AZFP(echodata: EchoData, user_cal_dict: dict) -> dict:
+def param2da(p_val: Union[int, float, list], channel: Union[list, xr.DataArray]) -> xr.DataArray:
     """
-    Get cal params using user inputs or values from data file.
+    Organize individual parameter in scalar or list to xr.DataArray with channel coordinate.
 
     Parameters
     ----------
-    echodata : EchoData
-        An EchoData object containing data to be calibrated
-    user_cal_dict : dict
-        A dictionary containing user-defined calibration parameters.
-        The user-defined calibration parameters will overwrite values in the data file.
+    p_val : int, float, or list
+        dictionary holding calibration params for one or more channels
+        each param has to be a scalar
+    channel : list or xr.DataArray
+        values to use for the output channel coordinate
 
     Returns
     -------
-    A dict containing the calibration parameters for the AZFP echosounder
+    an xr.DataArray with channel coordinate
     """
-    out_dict = dict.fromkeys(CAL_PARAMS["AZFP"])
-    if user_cal_dict is None:
-        user_cal_dict = {}
-
-    # Get params from Beam_group1
-    out_dict["equivalent_beam_angle"] = user_cal_dict.get(
-        "equivalent_beam_angle", echodata["Sonar/Beam_group1"]["equivalent_beam_angle"]
-    )
+    # TODO: allow passing in np.array as dict values to assemble a frequency-dependent cal da
 
-    # Get params from the Vendor_specific group
-    for p in ["EL", "DS", "TVR", "VTX", "Sv_offset"]:
-        # substitute if None in user input
-        out_dict[p] = user_cal_dict.get(p, echodata["Vendor_specific"][p])
+    if not isinstance(p_val, (int, float, list)):
+        raise ValueError("'p_val' needs to be one of type int, float, or list")
+    else:
+        if isinstance(p_val, list):
+            # Check length if p_val a list
+            if len(p_val) != len(channel):
+                raise ValueError("The lengths of 'p_val' and 'channel' should be identical")
 
-    return out_dict
+            return xr.DataArray(p_val, dims=["channel"], coords={"channel": channel})
+        else:
+            # if scalar, make a list to form data array
+            return xr.DataArray(
+                [p_val] * len(channel), dims=["channel"], coords={"channel": channel}
+            )
```

Comment:
```
    if not isinstance(p_val, (int, float, list)):
        raise ValueError("'p_val' needs to be one of type int, float, or list")
    
    if isinstance(p_val, list):
        # Check length if p_val a list
        if len(p_val) != len(channel):
            raise ValueError("The lengths of 'p_val' and 'channel' should be identical")

        return xr.DataArray(p_val, dims=["channel"], coords={"channel": channel})
    else:
        # if scalar, make a list to form data array
        return xr.DataArray(
            [p_val] * len(channel), dims=["channel"], coords={"channel": channel}
        )
```

The type checking already will raise the value error, so no need to have an else here for readability.

Reply:
Ah great, yes, I'll remove the else.